mix/CodeModel.cpp

```diff
@@ -213,11 +213,11 @@ void CodeModel::reset(QVariantMap const& _documents)
 	///@todo: cancel bg job
 	Guard l(x_contractMap);
 	releaseContracts();
-	Guard pl(x_pendingContracts);
-	m_pendingContracts.clear();
+	auto contracts = *m_pendingContracts;
```

Comment:
You can already use `.guard<GuardType>()`, I can add a non-templated function there.